src/utils/ERC1024.ts

```diff
@@ -0,0 +1,105 @@
+import type { Eip1024EncryptedData, Hex } from '@metamask/utils';
+import { base64ToBytes, bytesToBase64, hexToBytes } from '@metamask/utils';
+import { bytesToUtf8, utf8ToBytes } from '@noble/ciphers/utils';
+
+import { nacl } from './nacl';
+
+// ported from eth-sig-util
+const encrypt = (
+  receiverPublicKey: Hex | Uint8Array,
+  message: string,
+  version: string = 'x25519-xsalsa20-poly1305',
+): Eip1024EncryptedData => {
+  switch (version) {
+    case 'x25519-xsalsa20-poly1305': {
+      // generate ephemeral keypair
+      const ephemeralKeyPair = nacl.box.keyPair();
+
+      let publicKeyBytes;
+      // assemble encryption parameters
+      if (receiverPublicKey instanceof Uint8Array) {
+        publicKeyBytes = receiverPublicKey;
+      } else {
+        try {
+          publicKeyBytes = hexToBytes(receiverPublicKey);
+        } catch (error: any) {
+          throw new Error('Bad public key');
+        }
+      }
+
+      const messageBytes = utf8ToBytes(message);
+      const nonce = nacl.randomBytes(nacl.box.nonceLength);
+
+      // encrypt
+      const encryptedMessage = nacl.box.seal(
+        messageBytes,
+        nonce,
+        publicKeyBytes,
+        ephemeralKeyPair.secretKey,
+      );
+
+      // return encrypted data
+      return {
+        version: 'x25519-xsalsa20-poly1305',
+        nonce: bytesToBase64(nonce),
+        ephemPublicKey: bytesToBase64(ephemeralKeyPair.publicKey),
+        ciphertext: bytesToBase64(encryptedMessage),
+      } as Eip1024EncryptedData;
+    }
+    default:
+      throw new Error('Encryption type/version not supported');
```

Comment:
Since we are throwing, lets also append the version they were trying to encrypt with.

Reply:
Sure.
I mostly copied the code from eth-sig-util here, including error messages, to maintain compatibility, but since we're not talking about having drop-in replacements I can include your suggestion.

Reply:
Aha fair, just raising as it can be helpful for debugging.

This is a nit, so can be resolved otherwise.